src/arrayMethodSort.js

```diff
@@ -5,8 +5,93 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const arr = this;
+
+    // !!!bubble sort for dividing option!!!
+    if (compareFunction && arr.length > 1 && compareFunction(arr[0],
+      arr[1]) + arr[1] === arr[0]) {
+      for (let j = 0; j < arr.length; j++) {
+        for (let i = 0; i < arr.length - 1 - j; i++) {
+          if (compareFunction(arr[i], arr[i + 1]) > 0) {
+            const temp = arr[i];
+
+            arr[i] = arr[i + 1];
+
+            arr[i + 1] = temp;
+          }
+        }
+      }
+
+      return arr;
+    }
+
+    //  !!!selection sort for simple items!!!
+    // for (let i = 0; i < arr.length; i++) {
+    //   let min = arr[i];
+    //   let minIndex = i;
+
+    //   for (let j = i; j < arr.length; j++) {
+    //     const isMin = compareFunction
+    //       ? compareFunction(arr[j].toString(), min.toString())
+    //       : ((a, b) => a < b)(arr[j].toString(), min.toString());
+
+    //     if (isMin) {
+    //       min = arr[j];
+    //       minIndex = j;
+    //     }
+    //   }
+
+    //   const temp = arr[i];
+
+    //   arr[i] = arr[minIndex];
+
+    //   arr[minIndex] = temp;
+    // }
```

Comment:
Please send a file on review without unused or commented code. It will be a good practice